src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -21,19 +21,34 @@ typedef struct {
 } MPIDI_XPMEM_segmap_t;
 
 typedef struct MPIDI_XPMEM_seg {
-    off_t offset;               /* page aligned offset */
+    struct MPIDI_XPMEM_seg *parent;
+    struct MPIDI_XPMEM_seg *left;
+    struct MPIDI_XPMEM_seg *right;
+    uint64_t low;               /* page aligned low address of remote seg */
+    uint64_t high;              /* page aligned high address of remote seg */
     size_t size;                /* page aligned size */
-    void *vaddr;
+    void *vaddr;                /* virtual address attached in current process */
```

Comment:
I do not see the reason why you keep both `size` and `high`. If we just store `offset` and `size`, does it increase the overhead at performance-critical `search` routine (I consider `insert` is slow-path) ? If so, you can just store `low` and `high`; otherwise, only store `offset` and `size`.

Reply:
You do not answer my question:
> If we just store offset and size, does it increase the overhead at performance-critical search routine (I consider insert is slow-path) ?

Reply:
It does not affect `insert` process, but `search` process. If you only store `size` and `offset`, we just pay one addition overhead `high = size + offset`. I define `high` just for convenience.